File: LivrusVirtus/VCPP6/wxWin/wxODBC/ConEditoras.cpp
#include "resource.h"
#include "Consultas.h"
#include "Principal.h"
#include "ConsultasSQL.h"
#include "RotinasGlobais.h"
#include "ConEditoras.h"

BEGIN_EVENT_TABLE(CfrmConEditoras, CfrmConsultas)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNPESQUISAR, CfrmConEditoras::OnBtnPesquisar)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNLIMPAR, CfrmConEditoras::OnBtnLimpar)
  EVT_BUTTON(ID_FRMCONSULTAS_BTNFECHAR, CfrmConEditoras::OnBtnFechar)
  EVT_CLOSE(CfrmConEditoras::OnClose)
  EVT_GRID_SELECT_CELL(CfrmConEditoras::OnGridSelectCell)
  EVT_GRID_LABEL_LEFT_CLICK(CfrmConEditoras::OnGridLabelLeftClick)
END_EVENT_TABLE()

CfrmConEditoras::CfrmConEditoras(wxWindow *parent)
       : CfrmConsultas(parent) {
	IniciaControles();
}

CfrmConEditoras::~CfrmConEditoras() { }

void CfrmConEditoras::IniciaControles() {
 this->SetTitle(STR_CONEDITORA);
 this->labDescricao->SetLabel(wxT("Editora:  "));
 this->gridConsulta->SetColLabelValue(1, wxT("Editora"));
}

void CfrmConEditoras::OnClose(wxCloseEvent& event) {
  Fecha(MNU_CON_EDITORAS);
  event.Skip();
}

void CfrmConEditoras::OnBtnFechar(wxCommandEvent& event) {  
  Fecha(0);
  event.Skip();
}

void CfrmConEditoras::OnBtnLimpar(wxCommandEvent& event) {
  LimparDados();
  event.Skip();
}

void CfrmConEditoras::OnBtnPesquisar(wxCommandEvent& event) {
  PesquisarDados(
    ConsSQL->Editora('S',wxT(""), edtDescricao->GetValue()), STR_EDTNENC, 2);
  event.Skip();
}

void CfrmConEditoras::OnGridSelectCell(wxGridEvent& event) {
  setLabRegistros(event.GetRow());
  event.Skip();
}

void CfrmConEditoras::OnGridLabelLeftClick(wxGridEvent& event){
  setLabRegistros(event.GetRow());
  event.Skip();
}

File: LivrusVirtus/BoaConstructor/wxPython/ConEditoras.py
#Boa:Dialog:frmConEditoras

from wxPython.wx import *
from wxPython.grid import *
from Consultas import frmConsultas
from RotinasGlobais import RotinasGlobais
from ConsultasSQL import ConsultasSQL
from resource import Resource

def create(parent):
    return frmConEditoras(parent)

class frmConEditoras(frmConsultas):
    def _init_ctrls(self, prnt):
      frmConsultas._init_ctrls(self, prnt)
      self.SetTitle(Resource.STR_CONEDITORA)
      self.labDescricao.SetLabel("Editora:  ")
      self.gridConsulta.SetColLabelValue(1, "Editora")

      EVT_BUTTON(self.btnPesquisar, self.btnPesquisar.GetId(), 
        self.OnBtnPesquisarClick)
      EVT_BUTTON(self.btnLimpar, self.btnLimpar.GetId(), self.OnBtnLimparClick)
      EVT_BUTTON(self.btnFechar, self.btnFechar.GetId(), self.OnBtnFecharClick)
      EVT_CLOSE(self, self.OnFrmConEditorasClose)
      EVT_GRID_SELECT_CELL(self, self.OnGridSelectCell)
      EVT_GRID_LABEL_LEFT_CLICK(self, self.OnGridLabelLeftClick)

    def __init__(self, parent):
      self._init_ctrls(parent)

    def OnFrmConEditorasClose(self, event):
      self.Fecha(Resource.MNU_CON_EDITORAS)
      event.Skip()

    def OnBtnFecharClick(self, event):
      self.Fecha(0)
      event.Skip()

    def OnBtnLimparClick(self, event):
      self.LimparDados()
      event.Skip()

    def OnBtnPesquisarClick(self, event):
      self.PesquisarDados(
        ConsultasSQL.ConsSQL.Editora('S', "", self.edtDescricao.GetValue()),
        Resource.STR_EDTNENC, 2)
      event.Skip()

    def OnGridSelectCell(self, event):
      self.setLabRegistros(event.GetRow())
      event.Skip()

    def OnGridLabelLeftClick(self, event):
      self.setLabRegistros(event.GetRow())
      event.Skip()
